daemon/rpc_connect.go

```diff
@@ -235,6 +242,8 @@ func (r *RPC) connect(
 			event.Error = nil
 		}
 		r.events.Service.Connect.Publish(event)
+		r.connectionInfo.SetStatus(state.ConnectionStatus{State: pb.ConnectionState_DISCONNECTED, StartTime: nil})
```

Comment:
This call always goes before the event publishing. Maybe RPC does not need to know about `connectionInfo` and we can leave it for a subscriber to set the status properly based on received events?